src/scripts/main.js

```diff
@@ -3,7 +3,66 @@
 const calendar = document.querySelector('#calendar');
 
 function calendarTable(year, month, element) {
-  // WRITE YOUR CODE HERE
+  const date = new Date(year, month - 1);
+  const listOfDays = [];
+  let counter = 0;
+
+  function getFirstDay() {
+    const days = ['вс', 'пн', 'вт', 'ср', 'чт', 'пт', 'сб'];
+
+    return [days[date.getDay()], date.getDay()];
+  }
+
+  const numberOfDays = 32 - new Date(year, month - 1, 32).getDate();
+  const firstDay = getFirstDay(date);
+
+  for (let i = 0; i < firstDay[1]; i++) {
+    listOfDays.push('');
+  }
+
+  for (let i = 1; i <= numberOfDays; i++) {
+    listOfDays.push(i);
+  }
+
+  for (let i = 0; i <= 35 - listOfDays.length; i++) {
+    listOfDays.push('');
+  }
+
+  element.innerHTML = `
+    <table>
+      <tr>
+        <th>пн</th>
+        <th>вт</th>
+        <th>ср</th>
+        <th>чт</th>
+        <th>пт</th>
+        <th>сб</th>
+        <th>вс</th>
+      </tr>
+    </table>
+  `;
```

Comment:
если вверху вынести массив days в замыкание, то здесь можно дни недели записать через цикл
```suggestion
const rowHead = document.createElement('tr');

table.append(rowHead);

  for (const day of days) {
    const cell = document.createElement('th');
    
    rowHead.append(cell);        
  }
  `;
```